table/block_based/partitioned_filter_block.cc

```diff
@@ -484,21 +484,22 @@ Status PartitionedFilterBlockReader::CacheDependencies(const ReadOptions& ro,
       handle.offset() + handle.size() + BlockBasedTable::kBlockTrailerSize;
   uint64_t prefetch_len = last_off - prefetch_off;
   std::unique_ptr<FilePrefetchBuffer> prefetch_buffer;
-  rep->CreateFilePrefetchBuffer(
-      0, 0, &prefetch_buffer, false /* Implicit autoreadahead */,
-      0 /*num_reads_*/, 0 /*num_file_reads_for_auto_readahead*/);
-
-  IOOptions opts;
-  s = rep->file->PrepareIOOptions(ro, opts);
-  if (s.ok()) {
-    s = prefetch_buffer->Prefetch(opts, rep->file.get(), prefetch_off,
-                                  static_cast<size_t>(prefetch_len),
-                                  ro.rate_limiter_priority);
-  }
-  if (!s.ok()) {
-    return s;
+  if (tail_prefetch_buffer == nullptr || !tail_prefetch_buffer->Enabled()) {
+    rep->CreateFilePrefetchBuffer(
```

Comment:
Is this code still covered by unit tests? (I sometimes throw in an `assert(false);` to test that.)

Reply:
Yes - a couple tests did, including my TEST_P(PrefetchTest, BlockBasedTableTailPrefetch)